test/e2e/network/ingressclass.go

```diff
@@ -96,6 +97,38 @@ var _ = common.SIGDescribe("IngressClass [Feature:Ingress]", func() {
 		}
 	})
 
+	ginkgo.It("should allow IngressClass to have Namespace-scoped parameters [Serial]", func() {
+		ingressClass := &networkingv1.IngressClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ingressclass1",
+				Labels: map[string]string{
+					"ingressclass":  f.UniqueName,
+					"special-label": "generic",
+				},
+			},
+			Spec: networkingv1.IngressClassSpec{
+				Controller: "example.com/controller",
+				Parameters: &networkingv1.IngressClassParametersReference{
+					Scope:     utilpointer.StringPtr("Namespace"),
+					Namespace: utilpointer.StringPtr("foo-ns"),
+					Kind:      "fookind",
+					Name:      "fooname",
+					APIGroup:  utilpointer.StringPtr("example.com/v1"),
+				},
+			},
+		}
+		createdIngressClass, err := cs.NetworkingV1().IngressClasses().Create(context.TODO(), ingressClass, metav1.CreateOptions{})
+		if createdIngressClass.Spec.Parameters == nil {
+			framework.Failf("Expected IngressClass.spec.parameters to be set")
+		}
+		if createdIngressClass.Spec.Parameters.Scope == nil ||
+			*createdIngressClass.Spec.Parameters.Scope != "Namespace" {
+			framework.Failf("Expected IngressClass.spec.parameters.scope to be set to 'Namespace'")
```

Comment:
This could likely use a `got %v` extension - would also help clarify why this particular part of the test is failing.